### PR TITLE
Feature: C language under languages section

### LINKED ISSUE #639: [Feature Request] : Add C language under languages section
### Is your feature request related to a problem? Please describe.
I want to add C documentation under the language section .

### Describe the solution you'd like
The C documentation will include markdown files explaining key concepts i.e. Introduction , Variables , Operators etc..

**Would you like to work on this feature?**  
Yes

### PR DESCRIPTION
**Pull request by ishita-1305**
### Description
Added introduction to C, its data types, operators, decision-making, loop codes in the C folder under the languages section.

Fixes : #639 

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/languages/C/c-2.md [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+Relational operators compare two values and return a boolean result (`true` or `false`).
+
+| Operator | Description              | Example |
+| -------- | ------------------------ | ------- |
+| ==       | Equal to                 | x == y  |
+| !=       | Not equal to             | x != y  |
+| \>       | Greater than             | x \> y  |
+| \<       | Less than                | x \< y  |
+| \>=      | Greater than or equal to | x \>= y |
+| \<=      | Less than or equal to    | x \<= y |
```
Comment (ajay-dhangar) [inline docs/languages/C/c-2.md]: replace `<` with `&lt;` and `>` with `&gt;`

File: docs/languages/C/c-2.md [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+```
+
+## 3. Logical Operators
+
+Logical operators are used to perform logical operations and combine multiple conditions.
+
+| Operator | Description | Example             |
+| -------- | ----------- | ------------------- |
+| &&       | Logical AND | (x \> 5 && y \< 10) |
+| !        | Logical NOT | !(x \> 5)           |
```
Comment (ajay-dhangar) [inline docs/languages/C/c-2.md]: also, replace `<` and `>` because it's also introducing as a tags syntax

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
update your code based on suggestions

**Comment by ishita-1305** [comment]
@ajay-dhangar please review the changes

**Review by ajay-dhangar — APPROVED** [review]
Done

**Pull request merged**